pkg/api/store/clustertemplate/store.go

```diff
@@ -157,3 +169,35 @@ func isDefaultTemplateRevision(apiContext *types.APIContext, id string) (bool, e
 
 	return false, nil
 }
+
+func (p *Store) checkPermissionToCreateRevision(apiContext *types.APIContext, data map[string]interface{}) error {
+	userID := apiContext.Request.Header.Get("Impersonate-User")
+	if userID == "" {
+		return httperror.NewAPIError(httperror.NotFound, "invalid request: userID not found")
+	}
+
+	value, found := values.GetValue(data, managementv3.ClusterTemplateRevisionFieldClusterTemplateID)
+	if !found {
+		return httperror.NewAPIError(httperror.NotFound, "invalid request: clusterTemplateID not found")
+	}
+
+	clusterTemplateID := convert.ToString(value)
+	_, clusterTemplateName := ref.Parse(clusterTemplateID)
+	rbName, _ := rrbac.GetRoleNameAndVerbs(rrbac.ReadOnlyAccess, clusterTemplateName, rrbac.ClusterTemplateResource)
+	// check if rolebinding with this name exists and has the user (caller of this request) as a subject
+	rb, err := p.roleBindingLister.Get(namespace.GlobalNamespace, rbName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if rb == nil {
+		return nil
+	}
+
+	for _, sub := range rb.Subjects {
+		if sub.Name == userID {
```

Comment:
what happens if a template is public? It will have a read-only rolebinding created for *, what if a user tries to create a revision for such a template?